cscs-checks/prgenv/mpi_t.py

```diff
@@ -0,0 +1,64 @@
+import os
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+@rfm.simple_test
+class MpiTCheck(rfm.RegressionTest):
+    def __init__(self):
+        self.descr = 'Checks MPI_T control/performance variables/categories'
+        self.valid_systems = ['daint:gpu', 'dom:gpu', 'daint:mc', 'dom:mc']
+        self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-gnu', 'PrgEnv-pgi',
+                                    'PrgEnv-intel', 'PrgEnv-cray_classic']
+        self.build_system = 'SingleSource'
+        self.sourcepath = 'mpit_vars.c'
+        self.num_tasks_per_node = 1
+        self.variables = {'MPITEST_VERBOSE': '1', 'MPICH_VERSION_DISPLAY': '1'}
+        self.rpt = 'rpt'
+        self.executable_opts = ['&> %s' % self.rpt]
+        self.maintainers = ['JG']
+        self.tags = {'production', 'craype', 'maintenance'}
+
+    @rfm.run_before('sanity')
+    def set_sanity(self):
+        # 1/ MPI Control Variables:
+        # --- extract reference data:
+        regex = r'^(?P<vars>MPIR\S+)$'
+        ref = os.path.join(self.stagedir, 'mpit_control_vars.ref')
+        self.ref_control_vars = sorted(sn.extractall(regex, ref, 'vars'))
+        # --- extract runtime data:
+        regex = r'^\t(?P<vars>MPIR\S+)\t'
+        rpt = os.path.join(self.stagedir, self.rpt)
+        self.run_control_vars = sorted(sn.extractall(regex, rpt, 'vars'))
+        # 2/ MPI Performance Variables:
+        # --- extract reference data:
+        regex = r'(?P<vars>\w+)'
+        ref = os.path.join(self.stagedir, 'mpit_perf_vars.ref')
+        self.ref_perf_vars = sorted(sn.extractall(regex, ref, 'vars'))
+        # --- extract runtime data:
+        regex = r'^\t(?P<vars>(nem_|rma_)\S+)\t'
+        rpt = os.path.join(self.stagedir, self.rpt)
+        self.run_perf_vars = sorted(sn.extractall(regex, rpt, 'vars'))
+        # 3/ MPI Category:
+        # --- extract reference data:
+        regex = r'^(?P<category>.*)$'
+        ref = os.path.join(self.stagedir, 'mpit_categories.ref')
+        ref_cat_vars = sorted(sn.extractall(regex, ref, 'category'))
+        self.ref_cat_vars = list(filter(None, ref_cat_vars))
+        # --- extract runtime data:
+        regex = (r'^(?P<category>Category \w+ has \d+ control variables, \d+'
+                 r' performance variables, \d+ subcategories)')
+        rpt = os.path.join(self.stagedir, self.rpt)
+        self.run_cat_vars = sorted(sn.extractall(regex, rpt, 'category'))
+        # 4/ Extracted lists can be compared (when sorted):
+        self.sanity_patterns = sn.all([sn.assert_eq(self.ref_control_vars,
+                                                    self.run_control_vars),
+                                       sn.assert_eq(self.ref_perf_vars,
+                                                    self.run_perf_vars),
+                                       sn.assert_eq(self.ref_cat_vars,
+                                                    self.run_cat_vars),
+                                       ])
```

Comment:
Reformat as follows:

```suggestion
        self.sanity_patterns = sn.all([
            sn.assert_eq(self.ref_control_vars, self.run_control_vars),
            sn.assert_eq(self.ref_perf_vars, self.run_perf_vars),
            sn.assert_eq(self.ref_cat_vars, self.run_cat_vars),
        ])
```